src/bentoml/_internal/env_manager/manager.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import logging
+import subprocess
+from tempfile import NamedTemporaryFile
+from distutils.spawn import find_executable as which
+
+import fs
+import attrs
+from rich.status import Status
+
+from bentoml.exceptions import BentoMLException
+from bentoml._internal.configuration import get_debug_mode
+
+logger = logging.getLogger(__name__)
+is_debug_mode = get_debug_mode()
+
+if t.TYPE_CHECKING:
+    from fs.base import FS
+
+
+def decode(msg: bytes) -> str:
+    if msg:
+        return msg.decode("utf-8")
+    return ""
+
+
+def run_script_subprocess(
+    script_file_path: t.Union[str, os.PathLike[str]], capture_output: bool
+):
+    shell_path = which("bash")
+    if shell_path is None:
+        raise BentoMLException("Unable to locate a valid shell")
+
+    result = subprocess.run(
+        [shell_path, "-euxo", "pipefail", script_file_path],
+        capture_output=capture_output,
+    )
+    if result.returncode != 0:
+        if result.stderr and result.stdout:
+            logger.debug(decode(result.stdout))
+            logger.error(decode(result.stderr))
```

Comment:
I think that we should always log error no matter what.

Reply:
yep, changed